### ISSUE TITLE
Bazel fails to build a project which imports kubernetes-1.5

### ISSUE THREAD

**Issue by GinFungYJF** [kubernetes/kubernetes#42134]
My project imports kubernetes-1.5, and I uses bazel to build my project. I have do a local_repository to import kubernetes-1.5. When I build my project, the bazel complains 
```
# bazel build api-server
INFO: Found 1 target...
ERROR: /root/.cache/bazel/_bazel_root/688c88e95304704f955aa7cc70c4ad60/external/io_k8s_kubernetes/pkg/util/sets/BUILD:25:1: Executing genrule @io_k8s_kubernetes//pkg/util/sets:set-gen failed: linux-sandbox failed: error executing command /root/.cache/bazel/_bazel_root/688c88e95304704f955aa7cc70c4ad60/execroot/api-server/_bin/linux-sandbox ... (remaining 5 argument(s) skipped).
E0227 10:17:09.995985       3 main.go:51] Error: Failed making a parser: unable to add directory "./pkg/util/sets/types": unable to import "./pkg/util/sets/types": open /root/.cache/bazel/_bazel_root/688c88e95304704f955aa7cc70c4ad60/bazel-sandbox/c44f975b-9154-4587-b3d0-81526d369604-2/execroot/api-server/pkg/util/sets/types: no such file or directory
Target //:api-server failed to build
Use --verbose_failures to see the command lines of failed build steps.
ERROR: /var/paas/test1/cloud-fabric/src/api-server/BUILD:24:1 Executing genrule @io_k8s_kubernetes//pkg/util/sets:set-gen failed: linux-sandbox failed: error executing command /root/.cache/bazel/_bazel_root/688c88e95304704f955aa7cc70c4ad60/execroot/api-server/_bin/linux-sandbox ... (remaining 5 argument(s) skipped).
INFO: Elapsed time: 0.340s, Critical Path: 0.15s

```
the pkg/util/sets/BUILD in kubernetes-1.5:
```
package(default_visibility = ["//visibility:public"])

licenses(["notice"])

load(
    "@io_bazel_rules_go//go:def.bzl",
    "go_library",
    "go_test",
)

go_library(
    name = "go_default_library",
    srcs = [
        ":set-gen",
    ],
)

go_test(
    name = "go_default_test",
    srcs = ["set_test.go"],
    library = ":go_default_library",
    tags = ["automanaged"],
)

genrule(
    name = "set-gen",
    srcs = [
        "//pkg/util/sets/types:types.go",
        "//hack/boilerplate:boilerplate.go.txt",
    ],
    outs = [
        "byte.go",
        "doc.go",
        "empty.go",
        "int.go",
        "int64.go",
        "string.go",
    ],
    cmd = """
$(location //cmd/libs/go2idl/set-gen) \
    --input-dirs ./pkg/util/sets/types \
    --output-base $(GENDIR)/pkg/util \
    --go-header-file $(location //hack/boilerplate:boilerplate.go.txt) \
    --output-package sets
    """,
    tools = [
        "//cmd/libs/go2idl/set-gen",
    ],
)
```

What's more:
```
bazel version: 0.4.3
kubernetes: release-1.5
rules_go:0.3.2
```

**Comment by GinFungYJF** [comment]
@mikedanese

**Comment by mikedanese** [comment]
@ixdy @spxtr

**Comment by ixdy** [comment]
I think 1.6 is the first release in which we really support bazel. I don't think we cherry-picked any bazel fixes back to 1.5.

**Comment by mikedanese** [comment]
We should see if this is still broken in 1.6. I haven't tried importing k8s in a bazel workspace before.
